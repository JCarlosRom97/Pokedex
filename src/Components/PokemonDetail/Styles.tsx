import styled from "styled-components";
import { Link } from "react-router-dom";

export const PokemonDataContainer = styled.div`
    flex: 8;
    height: 100%;
    width: 100%;
    background: rgba(252, 252, 252, 0.7);
    padding: 30px;
`

export const BackButton = styled(Link)`
    text-decoration: none;
`

export const Text = styled.p<{align:string}>`
    text-align: ${(props) => props.align};
    font-weight: bold;
`

export const DataCointainer = styled.div`
    flex:1;
    flex-direction: row;
    display: flex;
`

export const Row = styled.div<{rows:number}>`
    flex:${(props) => props.rows}
`