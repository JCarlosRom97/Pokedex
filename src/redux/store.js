import { configureStore } from "@reduxjs/toolkit";
import PokemonReducer from "./reducers/index.reducer";
import detailReducer from "./reducers/detail.reducer";
export default configureStore({
    reducer: {
        PokemonDataList: PokemonReducer,
        detail: detailReducer
    },
})

