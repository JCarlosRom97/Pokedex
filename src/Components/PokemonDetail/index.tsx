import React from "react";
import { IDetailPokemonFetch } from "../../types";
import { PokemonDataContainer, Text, DataCointainer, Row, BackButton } from "./Styles";
import ProgressBar from "../ProgressBar";



const PokemonDetailColumn = ({data} : IDetailPokemonFetch) => {
    
    return(
        <PokemonDataContainer >
            <BackButton to={"/"}>Back</BackButton>
            <Text align="center">Type</Text>
            {data?.types?.map((type)=>(
                <Text key={type.type.name} align="center">{type.type.name}</Text>
            ))}

            <DataCointainer>
                <Row rows={3}>
                    <Text align="center">Number: {data.order}</Text>
                </Row>
                <Row rows={3}>
                    <Text align="center">Name: {data.name}</Text>
                </Row>
                <Row rows={3}>
                    <Text align="center">Height: {data.height}</Text>
                </Row>
                <Row rows={3}>
                    <Text align="center">Weigth: {data.weight}</Text>
                </Row>
            </DataCointainer>
            <Text align="left">Stats</Text>
            <DataCointainer>
                <Row rows={6}>
                    {data?.stats?.map((stats:{base_stat:number, stat:{name:string}})=>
                    (
                        <DataCointainer key={stats?.stat?.name}>
                            <Row rows={4}>
                                <Text align="left">{stats?.stat?.name}</Text>
                            </Row>
                            <Row rows={8}>
                                <ProgressBar bgcolor="red" completed={stats?.base_stat}/>
                            </Row>
                        </DataCointainer>
                    ))}
                </Row>
                <Row rows={4}>
                   <Text align="center">Abilities</Text>
                   {data?.abilities?.map((ability:{ability:{name:string}}) =>(
                        <Text key={ability?.ability?.name} align="center">{ability?.ability?.name}</Text>
                   ))}
                </Row>
            </DataCointainer>
        </PokemonDataContainer>
    )
}

export default PokemonDetailColumn;