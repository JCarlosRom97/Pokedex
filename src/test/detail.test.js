import renderer from "react-test-renderer"
import PokemonDetail from '../Detail/index';
import { Provider } from "react-redux";
import store from "../redux/store";
import * as axios from "axios";
import { BrowserRouter } from "react-router-dom";
import { render, waitFor } from "@testing-library/react";


jest.mock('axios');
jest.mock("react-router", () => ({
    ...jest.requireActual("react-router"),
    useLocation: () => ({
      state:{
        url: ''
      }
    })
  }));
describe('detail', () =>{
    test('snapshot should be the same', () =>{
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <PokemonDetail/>
                </BrowserRouter>
            </Provider>
        )
    
        let tree = component.toJSON();
    
        expect(tree).toMatchSnapshot()
    })

    test('renders users when API call succeeds', async () => {
        
     

        axios.get.mockResolvedValue({
            data: {   
                "types": [
                    {
                    "slot": 1,
                    "type": {
                        "name": "grass",
                        "url": "https://pokeapi.co/api/v2/type/12/"
                    }
                    },
                    {
                    "slot": 2,
                    "type": {
                        "name": "poison",
                        "url": "https://pokeapi.co/api/v2/type/4/"
                    }
                    }
                ],
                "order": 1,
                "name": "bulbasaur",
                "height": 7,
                "weight": 69,
                "stats": [
                    {
                    "base_stat": 45,
                    "effort": 0,
                    "stat": {
                        "name": "hp",
                        "url": "https://pokeapi.co/api/v2/stat/1/"
                    }
                    },
                    {
                    "base_stat": 49,
                    "effort": 0,
                    "stat": {
                        "name": "attack",
                        "url": "https://pokeapi.co/api/v2/stat/2/"
                    }
                    },
                    {
                    "base_stat": 49,
                    "effort": 0,
                    "stat": {
                        "name": "defense",
                        "url": "https://pokeapi.co/api/v2/stat/3/"
                    }
                    },
                    {
                    "base_stat": 65,
                    "effort": 1,
                    "stat": {
                        "name": "special-attack",
                        "url": "https://pokeapi.co/api/v2/stat/4/"
                    }
                    },
                    {
                    "base_stat": 65,
                    "effort": 0,
                    "stat": {
                        "name": "special-defense",
                        "url": "https://pokeapi.co/api/v2/stat/5/"
                    }
                    },
                    {
                    "base_stat": 45,
                    "effort": 0,
                    "stat": {
                        "name": "speed",
                        "url": "https://pokeapi.co/api/v2/stat/6/"
                    }
                    }
                ],
                "abilities": [
                    {
                    "ability": {
                        "name": "overgrow",
                        "url": "https://pokeapi.co/api/v2/ability/65/"
                    },
                    "is_hidden": false,
                    "slot": 1
                    },
                    {
                    "ability": {
                        "name": "chlorophyll",
                        "url": "https://pokeapi.co/api/v2/ability/34/"
                    },
                    "is_hidden": true,
                    "slot": 3
                    }
                ],
                "sprites": {
                    "front_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
                }
                
            }
        });
        

        const {container, asFragment} = render(
            <Provider store={store}>
                <BrowserRouter>
                    <PokemonDetail/>
                </BrowserRouter>
            </Provider>
        )

        const listNode = container.querySelector('div');
        expect(listNode.children).toHaveLength(2); 
    
    })

})