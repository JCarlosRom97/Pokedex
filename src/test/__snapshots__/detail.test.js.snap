// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detail renders users when API call succeeds 1`] = `
<DocumentFragment>
  <div
    class="sc-gtsqUy gBKKFr"
  >
    <div
      class="sc-hKFymg iQaxsz"
      columns="4"
    >
      <img
        class="sc-eCAqax wBXbV"
        src="pokeapi_256.png"
      />
      <img
        class="sc-eCAqax dvdsnT"
        height="300"
        paddingtop="100"
      />
    </div>
    <div
      class="sc-jSFipO iEmTfo"
    >
      <a
        class="sc-gKAaef gHRchn"
        href="/"
      >
        Back
      </a>
      <p
        align="center"
        class="sc-iCoHzw cHtrZT"
      >
        Type
      </p>
      <div
        class="sc-fujznN coqLWd"
      >
        <div
          class="sc-pNWRh bRebpx"
          rows="3"
        >
          <p
            align="center"
            class="sc-iCoHzw cHtrZT"
          >
            Number: 
          </p>
        </div>
        <div
          class="sc-pNWRh bRebpx"
          rows="3"
        >
          <p
            align="center"
            class="sc-iCoHzw cHtrZT"
          >
            Name: 
          </p>
        </div>
        <div
          class="sc-pNWRh bRebpx"
          rows="3"
        >
          <p
            align="center"
            class="sc-iCoHzw cHtrZT"
          >
            Height: 
          </p>
        </div>
        <div
          class="sc-pNWRh bRebpx"
          rows="3"
        >
          <p
            align="center"
            class="sc-iCoHzw cHtrZT"
          >
            Weigth: 
          </p>
        </div>
      </div>
      <p
        align="left"
        class="sc-iCoHzw gezCdT"
      >
        Stats
      </p>
      <div
        class="sc-fujznN coqLWd"
      >
        <div
          class="sc-pNWRh bRebrs"
          rows="6"
        />
        <div
          class="sc-pNWRh bRebsC"
          rows="4"
        >
          <p
            align="center"
            class="sc-iCoHzw cHtrZT"
          >
            Abilities
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`detail snapshot should be the same 1`] = `
<div
  className="sc-gtsqUy gBKKFr"
>
  <div
    className="sc-hKFymg iQaxsz"
    columns={4}
  >
    <img
      className="sc-eCAqax wBXbV"
      src="pokeapi_256.png"
    />
    <img
      className="sc-eCAqax dvdsnT"
      height={300}
      paddingtop={100}
    />
  </div>
  <div
    className="sc-jSFipO iEmTfo"
  >
    <a
      className="sc-gKAaef gHRchn"
      href="/"
      onClick={[Function]}
    >
      Back
    </a>
    <p
      align="center"
      className="sc-iCoHzw cHtrZT"
    >
      Type
    </p>
    <div
      className="sc-fujznN coqLWd"
    >
      <div
        className="sc-pNWRh bRebpx"
        rows={3}
      >
        <p
          align="center"
          className="sc-iCoHzw cHtrZT"
        >
          Number: 
        </p>
      </div>
      <div
        className="sc-pNWRh bRebpx"
        rows={3}
      >
        <p
          align="center"
          className="sc-iCoHzw cHtrZT"
        >
          Name: 
        </p>
      </div>
      <div
        className="sc-pNWRh bRebpx"
        rows={3}
      >
        <p
          align="center"
          className="sc-iCoHzw cHtrZT"
        >
          Height: 
        </p>
      </div>
      <div
        className="sc-pNWRh bRebpx"
        rows={3}
      >
        <p
          align="center"
          className="sc-iCoHzw cHtrZT"
        >
          Weigth: 
        </p>
      </div>
    </div>
    <p
      align="left"
      className="sc-iCoHzw gezCdT"
    >
      Stats
    </p>
    <div
      className="sc-fujznN coqLWd"
    >
      <div
        className="sc-pNWRh bRebrs"
        rows={6}
      />
      <div
        className="sc-pNWRh bRebsC"
        rows={4}
      >
        <p
          align="center"
          className="sc-iCoHzw cHtrZT"
        >
          Abilities
        </p>
      </div>
    </div>
  </div>
</div>
`;
